packages/jest-types/__typetests__/jest.test.ts

```diff
@@ -98,6 +98,9 @@ expectError(jest.enableAutomock('moduleName'));
 expectType<typeof jest>(jest.isolateModules(() => {}));
 expectError(jest.isolateModules());
 
+expectType<typeof jest>(jest.isolateModulesAsync(() => Promise.resolve(jest.fn())));
```

Comment:
The type tests will just type checked, but not executed. Programatic API of the TS compiler is used for that.

To run the type tests you should build the repo and run `yarn test-types`. Remember to rebuild after making changes in the code base (changes in test file does not need to be build).

```suggestion
expectType<Promise<typeof jest>>(jest.isolateModulesAsync(async () => {}));
```